app/uploaders/simple_forms_api/form_remediation/uploader.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+require 'simple_forms_api/form_remediation/configuration/base'
+
+module SimpleFormsApi
+  module FormRemediation
+    class Uploader < CarrierWave::Uploader::Base
+      include UploaderVirusScan
+
+      class << self
+        def s3_settings
+          config = Configuration::Base.new
+          config.s3_settings
+        end
+
+        def new_s3_resource
+          client = Aws::S3::Client.new(region: s3_settings.region)
+          Aws::S3::Resource.new(client:)
+        end
+
+        def get_s3_link(file_path)
+          new_s3_resource.bucket(s3_settings.bucket)
+                         .object(file_path)
+                         .presigned_url(:get, expires_in: 30.minutes.to_i)
+        end
+      end
+
+      def size_range
+        (1.byte)...(150.megabytes)
+      end
+
+      # Allowed file types, including those specific to benefits intake
+      def extension_allowlist
+        %w[bmp csv gif jpeg jpg json pdf png tif tiff txt zip]
+      end
+
+      def initialize(directory:, config: Configuration::Base.new)
+        raise 'The S3 directory is missing.' if directory.blank?
+        raise 'The configuration is missing.' unless config
+
+        @config = config
+        @directory = directory
+
+        super()
+        set_storage_options!
+      end
+
+      def store_dir
+        @directory
```

Comment:
So does this store the directory just on `initialize` or when `store_dir` is invoked?

Reply:
`store_dir` is short for the directory where storing will take place. So this variable is set/referenced when the instance invokes it but the directory will have already been stored upon initialization.